modules/core/src/smithy4s/StreamingSchema.scala

```diff
@@ -16,7 +16,18 @@
 
 package smithy4s
 
-sealed trait StreamingSchema[A]
+sealed trait StreamingSchema[A] { self =>
+
+  def transformHintsTransitively(f: Hints => Hints): StreamingSchema[A] =
+    self match {
+      case StreamingSchema.Streamed(fieldName, schema) =>
+        StreamingSchema.Streamed(
+          fieldName,
+          schema.transformHintsTransitively(f)
+        )
+      case _ => self
```

Comment:
I think this should explicitly match against `NoStream` to avoid accidentally missing other cases if there are any in the future

Reply:
I don't think we can do that because StreamingSchema is invariant in A

Reply:
yeah, that's a bit messed up in scala 2.

This compiles for me on all Scala versions:

```suggestion
      case _: StreamingSchema.NoStream.type => self
```